### PR TITLE
issue: 1670999 Print mlx4 steering warning in ib_ctx_handler_collection

### PR DESCRIPTION
**Pull request by liranoz12**
VMA provides a warning while mlx4-steering-rules are disabled.
This warning is presented during main initialization even when no mlx4
devices are available and no socket-api functions were used.

Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

File: src/vma/dev/net_device_val.cpp [hunk 1]
```diff
@@ -1567,7 +1567,7 @@ bool net_device_val::verify_enable_ipoib(const char* interface_name)
 #ifndef DEFINED_IBV_QP_INIT_SOURCE_QPN
 	// Dont offload mlx5 devices if source qpn is not defined.
 	ib_ctx_handler* ib_ctx = g_p_ib_ctx_handler_collection->get_ib_ctx(get_ifname_link());
-	if (!strncmp(ib_ctx->get_ibname(), "mlx5", 4)) {
+	if (!ib_ctx->is_mlx4()) {
```
Comment (igor-ivanov) [inline src/vma/dev/net_device_val.cpp]: It is dangerous way to check. Let imagine that there is more than just mlx4 and mlx5. Please consider `is_mlx5()` to be consistent.

File: src/vma/dev/net_device_val.cpp [hunk 2]
```diff
@@ -1675,7 +1675,7 @@ bool net_device_val::verify_qp_creation(const char* ifname, enum ibv_qp_type qp_
 	qp_init_attr.send_cq = cq;
 
 	// Set source qpn for mlx5 IPoIB devices
-	if (qp_type == IBV_QPT_UD && !strncmp(p_ib_ctx->get_ibname(), "mlx5", 4)) {
+	if (qp_type == IBV_QPT_UD && !p_ib_ctx->is_mlx4()) {
```
Comment (igor-ivanov) [inline src/vma/dev/net_device_val.cpp]: It is dangerous way to check. Let imagine that there is more than just mlx4 and mlx5. Please consider `is_mlx5()` to be consistent.

[the remark this answers is not in the document]
Comment (liranoz12) [reply src/vma/dev/net_device_val.cpp]: :+1:

File: src/vma/dev/net_device_val.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (liranoz12) [reply src/vma/dev/net_device_val.cpp]: +1

**Pull request merged**